### DISCUSSION TITLE
'subscribeKey' gets both the new value and the old value

### DISCUSSION THREAD

**Discussion by ayangweb** [pmndrs/valtio#907] [Ideas]
Perhaps `subscribeKey` can be converted to do something similar to vue's `watch`?

```js
subscribeKey(xxx, "xxx", (value, oldValue) => {
	// do something
});
```

**Comment by dai-shi** [comment]
Thanks for the suggestion.
I have kind of mixed feeling. For one, I rejected adding features like #887.
For another, it's trivial to implement.
But, the fact that `subscribeKey` is to similar to `subscribe` which doesn't have the prev value pushes me back.

So, unless we have a huge demand of this feature, I think leaving it as is is reasonable.
And, if we have a huge demand, we should probably explore a new library in valtiojs/* including `subscribeKeys`.

**Reply by ayangweb** [reply]
Okay, thank you.

**Reply by ayangweb** [reply]
So if you want to add an `immediate` similar to vue's `watch` method, also want to implement it in valtiojs/*?

**Reply by dai-shi** [reply]
https://vuejs.org/guide/essentials/watchers#eager-watchers
Is it about eager callback?
I think it's `sync` or `notifyInSync` for us.
https://github.com/pmndrs/valtio?tab=readme-ov-file#update-synchronously
https://github.com/pmndrs/valtio/blob/c1a18e214abb4d894870d3c7160a5fa226bdb4b8/src/vanilla/utils/subscribeKey.ts#L8

**Reply by ayangweb** [reply]
What I'm trying to say is that the page can trigger the `subscribeKey` callback function the first time it's opened, whether the value has changed or not

**Reply by dai-shi** [reply]
Oh, it's probably something I call "fireImmediately". Yeah, it might be nice to support is in your new library.

FYI, this is what I would do:
```js
const callback = () => ...;
subscribe(proxy, callback);
callback(); // fire immediately
```

**Reply by ayangweb** [reply]
Looking forward to this feature 🥳
